compiler/src/dotty/tools/dotc/transform/ElimJavaPackages.scala

```diff
@@ -0,0 +1,46 @@
+package dotty.tools.dotc.transform
+
+import dotty.tools.dotc.ast.tpd._
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Flags._
+import dotty.tools.dotc.core.Types.{Type, TypeRef}
+import dotty.tools.dotc.transform.TreeTransforms.{MiniPhaseTransform, TransformerInfo}
+
+/**
+  * Eliminates syntactic references to Java packages, so that there's no chance
+  * they accidentally end up in the backend.
+  */
+class ElimJavaPackages extends MiniPhaseTransform {
```

Comment:
> InfoTransformer doesn't apply, because the type of the tree

DenotTransformer are required when you change "type of a name", i.e. Denotation.
E.g. if a method/field used to return one type, but now returns another. Or if it takes more agruments. Or if a field became a method. Or if new methods\fields\types\classes are introduced.

You don't need to extend DenotTransformer here if are rebuilding trees that won't affect typing of "names". This phase doesn't.